R/generateIIASASubmission.R

```diff
@@ -97,82 +131,74 @@ generateIIASASubmission <- function(mifs = ".", mapping = NULL, model = NULL, #
            " before 1.111.0 on 2023-05-26, please use piamInterfaces version 0.9.0 or earlier, see PR #128.")
   }
 
-  # generate mapping file, if it doesn't exist yet
-  if (length(mapping) > 0 || is.null(mappingFile)) {
-    mapData <- generateMappingfile(templates = mapping, outputDirectory = NULL,
-                                   fileName = NULL, model = model, logFile = logFile,
-                                   iiasatemplate = iiasatemplate)[["mappings"]]
-    if (! is.null(mappingFile)) {
-      warning("mappingFile='", mappingFile, "' is ignored because you specified mapping.")
-      mappingFile <- NULL
-    }
-  } else if (length(mappingFile) == 1 && file.exists(mappingFile)) {
-    mapData <- read.csv2(mappingFile)
-  } else {
-    stop("Specify either one single 'mappingFile' or use (multiple) templates using the 'mapping' parameter.")
-  }
+  mifdata <- .setModelAndScenario(mifdata, model, removeFromScen, addToScen)
 
-  mapData <- mapData %>%
-    mutate(
-      !!sym("factor") := as.numeric(!!sym("factor")),
-      # this is not optimal and error-prone: we must dissect variable into variable and unit again
-      # could be avoided, if we expect mappings with variable and unit fields
-      # instead of having the unit as part of the variable name
-      !!sym("piam_unit") := unitsplit(!!sym("piam_variable"))$unit, # nolint
-      !!sym("piam_variable") := removePlus(unitsplit(!!sym("piam_variable"))$variable), # nolint
-      !!sym("Unit") := unitsplit(!!sym("Variable"))$unit, # nolint
-      !!sym("Variable") := unitsplit(!!sym("Variable"))$variable # nolint
-    )
-
-  message("\n### Generating submission file using mapping ", paste(c(mapping, mappingFile), collapse = ", "), ".")
+  # generate mapping from templates ----
 
-  mifdata <- .setModelAndScenario(mifdata, model, removeFromScen, addToScen)
+  mapData <- NULL
+
+  for (i in seq_along(templates)) {
+    t <- getTemplate(templates[i]) %>%
+      filter(.data$piam_variable != "", !is.na(.data$piam_variable), .data$piam_variable != "TODO") %>%
+      mutate(
+        "piam_variable" = removePlus(.data$piam_variable),
+        "piam_factor" = ifelse(is.na(.data$piam_factor), 1, as.numeric(.data$piam_factor))
+      ) %>%
+      select("Variable", "Unit", "piam_variable", "piam_unit", "piam_factor")
```

Comment:
Not sure, but: shouldn't we simply go to lowercase in the templates as well? Meaning converting all colnames to lowercase here and then using `variable` and `unit` consistently? It always confused me that it was `piam_variable` but then `Variable`.

Reply:
done